sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -2628,7 +2633,7 @@ public void populateDisplayData(DisplayData.Builder builder) {
     public TableReference getOrCreateTable(BigQueryOptions options, String tableSpec)
         throws IOException {
       TableReference tableReference = parseTableSpec(tableSpec);
-      if (!createdTables.contains(tableSpec)) {
+      if (!createdTables.contains(tableSpec) && jsonTableSchema != null) {
```

Comment:
Instead of using `jsonTableSchema != null` can we check `CreateDisposition == NEVER`?